packages/arb-token-bridge-ui/src/hooks/TransferPanel/useGasSummary.ts

```diff
@@ -0,0 +1,252 @@
+      address: walletAddress,
+      l2Provider: childChainProvider
+    }
+
+    let estimateGasResult: GasEstimationResult = INITIAL_GAS_ESTIMATION_RESULT
+
+    try {
+      setGasSummaryStatus('loading')
+
+      if (isDepositMode) {
+        estimateGasResult = token
+          ? await depositTokenEstimateGas({
+              ...estimateGasFunctionParams,
+              l1Provider: parentChainProvider,
+              erc20L1Address: token.address
+            })
+          : await depositEthEstimateGas({
+              ...estimateGasFunctionParams,
+              l1Provider: parentChainProvider
+            })
+      } else {
+        const partialEstimateGasResult = await withdrawInitTxEstimateGas({
+          ...estimateGasFunctionParams,
+          erc20L1Address: token ? token.address : undefined
+        })
+
+        estimateGasResult = {
+          ...partialEstimateGasResult,
+          estimatedL2SubmissionCost: constants.Zero
+        }
+      }
+
+      setResult(estimateGasResult)
+      setGasSummaryStatus('success')
+    } catch (error) {
+      console.error(error)
+      setGasSummaryStatus('error')
+    }
+  }, [
+    // Re-run gas estimation when:
+    isDepositMode, // when user switches deposit/withdraw mode
+    amountDebounced,
+    token, // when the token changes
+    walletAddress, // when user switches account or if user is not connected
+    setGasSummaryStatus,
+    parentChainProvider,
+    childChainProvider
+  ])
+
+  // Estimated L1 gas fees, denominated in Ether, represented as a floating point number
+  const estimatedL1GasFees = useMemo(() => {
+    const gasPrice = isDepositMode ? parentChainGasPrice : childChainGasPrice
+    return parseFloat(utils.formatEther(result.estimatedL1Gas.mul(gasPrice)))
+  }, [
+    isDepositMode,
+    parentChainGasPrice,
+    childChainGasPrice,
+    result.estimatedL1Gas
+  ])
+
+  // Estimated L2 gas fees, denominated in Ether, represented as a floating point number
+  const estimatedL2GasFees = useMemo(
+    () =>
+      parseFloat(
+        utils.formatEther(
+          result.estimatedL2Gas
+            .mul(childChainGasPrice)
+            .add(result.estimatedL2SubmissionCost)
+        )
+      ),
+    [
+      result.estimatedL2Gas,
+      childChainGasPrice,
+      result.estimatedL2SubmissionCost
+    ]
+  )
+
+  useEffect(() => {
+    // Since we are using a debounced value, it's possible for the value to be outdated
+    // Wait for it to sync before running the gas estimation
+    if (!amountDebounced.eq(amountBigNumber)) {
+      setGasSummaryStatus('loading')
+      return
+    }
+
+    if (!isDepositMode) {
+      if (
+        isTokenArbitrumOneNativeUSDC(token?.address) ||
+        isTokenArbitrumSepoliaNativeUSDC(token?.address)
+      ) {
+        setGasSummaryStatus('unavailable')
+        return
+      }
+    }
+
+    if (
+      arbTokenBridgeLoaded &&
+      arbTokenBridge &&
+      arbTokenBridge.eth &&
+      arbTokenBridge.token
+    ) {
```

Comment:
We might be able to just kill all of this :hocho:

Reply:
addressed here https://github.com/OffchainLabs/arbitrum-token-bridge/pull/1490

Reply:
This [PR](https://github.com/OffchainLabs/arbitrum-token-bridge/pull/1479) should fix all those check